lib/cloudregister/registerutils.py

```diff
@@ -531,40 +530,187 @@ def setup_registry(registry_fqdn, username, password):
 
 
 # ----------------------------------------------------------------------------
-def set_registry_auth_token(registry_fqdn, username, password):
-    """Set the auth token to pull images from SUSE registry."""
-    auth_token = base64.b64encode('{username}:{password}'.format(
-        username=username,
-        password=password
-    ).encode()).decode()
-    registry_credentials = {registry_fqdn: {'auth': auth_token}}
+def get_registry_credentials(set_new):
+    """Read the registry credentials file
+       and return its content or an empty dict."""
     config_json = {}
+    failed = None
+    file_error = True
     try:
         with open(REGISTRY_CREDENTIALS_PATH, 'r') as cred_json:
             config_json = json.load(cred_json)
-        # set the new registry credentials,
-        # independently of what that content was
-        config_json['auths'].update(registry_credentials)
-    except (FileNotFoundError, KeyError):
-        # file does not exist or "auths" key is not set
-        config_json.update({'auths': registry_credentials})
+    except OSError as error:
+        logging.info(str(error))
+        action = 'open'
     except json.decoder.JSONDecodeError:
-        logging.info(
-            'Error found when opening %s' % REGISTRY_CREDENTIALS_PATH
+        action = 'parse'
+    else:
+        file_error = False
+
+    if file_error:
+        error_msg = 'Unable to %s existing %s, preserving file as %s.bak' % (
+            action, REGISTRY_CREDENTIALS_PATH, REGISTRY_CREDENTIALS_PATH
         )
-        config_json.update({'auths': registry_credentials})
+        if set_new:
+            error_msg += ', writing new credentials'
+        logging.info(error_msg)
+        mv_file_cmd = 'mv -Z {} {}.bak'.format(
+            REGISTRY_CREDENTIALS_PATH, REGISTRY_CREDENTIALS_PATH
+        ).split()
+        failed = exec_subprocess(mv_file_cmd)
+        message = 'File not preserved.' if failed else 'File preserved.'
+        logging.info(message)
 
+    return config_json, failed
```

Comment:
Since `failed` has 3 states, which does not make sense for a variable named `failed` we should rename it to `status`.